### DISCUSSION TITLE
How to cache content in Next.js

### DISCUSSION THREAD

**Discussion by carlesandres** [supabase/supabase#18551] [Questions]
When using supabase with Next.js app router, I would like to have some components be server-rendered at build time.

I am currently using `@supabase/auth-helpers-nextjs` which allows me to create a supabase client but it requires a `cookies` parameter. This requirement makes each component that uses it a `dynamic` component. That prevents the component from being statically built (and therefore cached).

However, there are many occasions in which the contents I need to fetch from Supabase are not user-specific and I could benefit massively from using Next.js caching mechanisms by querying the database anonymously and pre-rendering some components for all my users.

Is there a way to achieve this using a Supabase library? How are people addressing this scenario? Is Supabase suggesting an approach to this?

**Comment by ailon** [comment]
In those "anonymous" parts you should be able to use the Supabase client from the `supabase-js` and not from `auth-helpers` and that wouldn't require cookies.
